### ISSUE TITLE
[Meta][Lens migration] Phases and Issues tracker

### ISSUE THREAD

**Issue by milan-elastic** [elastic/integrations#6489]
This meta issue is related to track the lens migration phases and issues that raised as part of the Lens migration activity. 

## Meta/Phases

- [x] Lens Migration phase 1 (OBS Integrations) -  #5221 
- [x] Lens Migration  phase 2 (AWS integrations) - #6141 
- [x] Lens Migration phase 3 (AWS, Azure, GCP integrations) - #6907
- [x]  Lens Migration phase 4 (Cloud Native Integration) - #7797
- [x]  Lens Migration phase 5 (Last Phase) - #7815 
## Open Kibana/Lens Issues
<html>
<body>
<google-sheets-html-origin>

Issue | Affected Integrations | Summary | Severity | Next step | Proposed Workaround | Phase
-- | -- | -- | -- | -- | -- | --
[kibana#135883](https://github.com/elastic/kibana/issues/135883) | STAN, NATS |  [Lens] Add IP range function |  Critical | Provided workaround as per the Kibana team [suggestions](https://github.com/elastic/kibana/issues/135883#issuecomment-1785512040) | Yes | [Phase-4](https://github.com/elastic/integrations/issues/7797)
[kibana#100231](https://github.com/elastic/kibana/issues/100231) | AWS ELB | Trim axis labels/ticks for long text field values | Blocker | Once Kibana team will resolve this issue, we can proceed further. | **Yes** - Create table chart to display top user agents[[Reference](https://github.com/elastic/integrations/issues/6201#issuecomment-1580736074)] | [Phase-2](https://github.com/elastic/integrations/issues/6141)
[kibana#72063](https://github.com/elastic/kibana/issues/72063) | ActiveMQ, Apache, Redis | Splitting a chart into multiple smaller charts by a breakdown dimension is not supported in the lens. | Critical | - | **Yes**  - As per this [comment](https://github.com/elastic/integrations/pull/5693#issuecomment-1484881970), we have converted tag cloud visualization to horizontal bar chart | [Phase-1](https://github.com/elastic/integrations/issues/5221)
[kibana#153010](https://github.com/elastic/kibana/issues/153010) | Apache Spark | Facing issue related to aggregation based gauge chart to lens gauge chart while migrating. | Major | - | **Yes**  - As [discussed](https://github.com/elastic/integrations/issues/5674#issuecomment-1482736259) with @SubhrataK,, we can convert gauge charts to metric. Also Kibana team will work on gauge charts as per this [comment](https://github.com/elastic/kibana/issues/153010#issuecomment-1480943098). | [Phase-1](https://github.com/elastic/integrations/issues/5221)
[kibana#153324](https://github.com/elastic/kibana/issues/153324) | Kafka  | Support multiple metrics in one chart using lens | Minor | - | **Yes**  - Converted into single-single metric chart | [Phase-1](https://github.com/elastic/integrations/issues/5221)

</body>
</html>

## Completed Kibana/Lens Issues
<html>
<body>
<google-sheets-html-origin>

Issue | Affected Integrations | Summary | Severity | Next step | Proposed Workaround | Phase
-- | -- | -- | -- | -- | -- | --
[kibana#95542](https://github.com/elastic/kibana/issues/95542) | ActiveMQ, Oracle WebLogic | The TagCloud visualizations are not supported in the lens. | Critical | - | **Yes**  - As per this [comment](https://github.com/elastic/integrations/pull/5250#:~:text=issues/95542%5D-,Yes%2C%20we%20suggest%20horizontal%20bar%20chart%20as%20a%20current%20alternative.,-Manually%20handle%20timestamp), Converted TagCloud to Horizontal bar chart | [Phase-1](https://github.com/elastic/integrations/issues/5221)
[kibana#96576](https://github.com/elastic/kibana/issues/96576) | IBM MQ | The text type field is not supported in table type visualizations of Lens | Blocker | Once Kibana team will resolve this issue, we can proceed further. | No (As [discussed](https://github.com/elastic/integrations/pull/5495#issuecomment-1477590336)  with @agithomas, removing the text field is not the idle option. Hence we have kept the visualization as it is.) | [Phase-1](https://github.com/elastic/integrations/issues/5221)
[Kibana#153378](https://github.com/elastic/kibana/issues/153378) | Apache, Redis, Hadoop | Not able to convert value into human readable format (Seconds / Minutes / Hours) | Blocker | Kibana team has provided the [workaround](https://github.com/elastic/kibana/issues/153378#issuecomment-1479731812). We tried the solution and discussed it with @SubhrataK but it was not user-friendly. So we have provided this [comment](https://github.com/elastic/kibana/issues/153378#issuecomment-1484867396) and asked the Kibana team to provide this feature in the Lens. | No | [Phase-1](https://github.com/elastic/integrations/issues/5221)
[kibana#72711](https://github.com/elastic/kibana/issues/72711) | MongoDB, Oracle | Lens doesn't support number formatting. (For example 20000 -> 20k) | Blocker | Waiting on Kibana team to resolve this issue, Once it gets resolved then we can proceed further. | No | [Phase-1](https://github.com/elastic/integrations/issues/5221)
[kibana#153326](https://github.com/elastic/kibana/issues/153326) | Kafka, HAProxy, Apache | Not having fetaure to set last bucket value interval as auto | Critical | - | **Yes**  - The metrics chart needs to convert into the lens. If metrics charts contains differences which uses the date histogram we can convert them to line charts. [Link](https://github.com/elastic/integrations/pull/5636#issuecomment-1491731261) | [Phase-1](https://github.com/elastic/integrations/issues/5221)
[elastic-charts#135](https://github.com/elastic/elastic-charts/issues/135) | ActiveMQ, Zookeeper   | TSVB visualization doesn’t show negative values in UI. But the lens shows both negative and positive values. (This will be fixed in 8.8) | Minor | - | Will be fixed in 8.8 | [Phase-1](https://github.com/elastic/integrations/issues/5221)

</body>
</html>

## Future/Backlog Work (Integration issues that found during the migrations):

- **Blocked issues: Visualizations that are not migrated because no alternatives exist** 
<html>
<body>
<google-sheets-html-origin>

Integration | Type | Summary | Severity | Status | Next step | Phase
-- | -- | -- | -- | -- | -- | --
IBM MQ - #5662 | Enhancement | Migrate Top 5 Errors [Logs IBM MQ] panel to the lens | Critical | Open | Kibana team will support text field after that we can take this up: - [link](https://github.com/elastic/kibana/issues/96576#issuecomment-815878764) | [Phase-1](https://github.com/elastic/integrations/issues/5221)

- **Blocked issues: Visualizations that are migrated with work around but still need a future work**
NA

- **Unblocked issues: Visualizations that got migrated with work-around but still need a future work**
<html>
<body>
<google-sheets-html-origin>

Integration | Type | Summary | Severity | Status | Next step | Phase
-- | -- | -- | -- | -- | -- | --
Azure Metrics - #7585 |  Enhancement | Add [Azure Metrics] suffix to the panels titles | Minor  | Open | Need to update visualizations | [Phase 3](https://github.com/elastic/integrations/issues/6907)
MongoDB - #7116  | Enhancement | Add custom number formatting to visualizations | Critical | Open | Kibana team will support custom format in lens after that we can take this up: - [link](https://github.com/elastic/kibana/issues/72711) | [Phase-1](https://github.com/elastic/integrations/issues/5221)
AWS Billing - #6835 | Enhancement | Add custom format for Total Estimated Charges visualization | Major | Open | Custom field format supported in 8.9 | [Phase 3](https://github.com/elastic/integrations/issues/6907)
GCP Billing - #7252 | Enhancement | Convert legacy metrics to new lens metric | Major | Open | Update legacy metric visualization | [Phase 3](https://github.com/elastic/integrations/issues/6907)
AWS Security Hub - #6864 | Enhancement | Need to add title in AWS Security Hub dashboard panels | Minor | Open | Update panel titles | [Phase 3](https://github.com/elastic/integrations/issues/6907)
IIS - #5743 | Enhancement | Revisit the color scheme in IIS dashboards panels | Minor | Open | Need to revisit this issue as per this [comment](https://github.com/elastic/integrations/pull/5727#issuecomment-1488616000) | [Phase-1](https://github.com/elastic/integrations/issues/5221)

- **Completed issues**
<html>
<body>
<google-sheets-html-origin>

Integration | Type | Summary | Severity | Status | Next step | Phase
-- | -- | -- | -- | -- | -- | --
~AWS Security Hub - #6927~ | ~Enhancement~ | ~Update donut charts with pie/treemap~ | ~Minor~ | ~Open~ | ~Update donut charts~ | ~[Phase 3](https://github.com/elastic/integrations/issues/6907)~
~AWS Redshift - #6909~ | ~Enhancement~ | ~Convert legacy metrics to new lens metric~ | ~Major~ | ~Open~ | ~Update legacy metric visualization~ | ~[Phase 3](https://github.com/elastic/integrations/issues/6907)~
~Oracle - #7117~  | ~Enhancement | Add custom number formatting to visualizations~ | ~Critical~ | ~Open~ | ~Kibana team will support custom format in lens after that we can take this up: - [link](https://github.com/elastic/kibana/issues/72711)~ | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~HAProxy - #5584~ | ~Enhancement~ | ~Migrate visualization to lens~ | ~Critical~ | ~In Review~ | ~Blocked on 8.6.0 dependency. https://github.com/elastic/integrations/pull/5611#issuecomment-1501731614~ | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~Azure Application Insights - https://github.com/elastic/integrations/issues/7231~ | ~Enhancement~ | ~Migrate visualization to lens~ | ~Critical~ | ~Closed~ | ~-~  | ~[Phase 3](https://github.com/elastic/integrations/issues/6907)~
~~NATS , STAN- https://github.com/elastic/integrations/issues/8021~~ | ~~Enhancement~~ | ~~Migrate Client IP Count Timeline to lens~~ | ~~Critical~~ | ~~Open~~ | ~~Blocked on Add IP range function https://github.com/elastic/kibana/issues/135883~~ | ~~[Phase-4](https://github.com/elastic/integrations/issues/7797)~~
~~Hadoop - #7112~~  | ~~Enhancement~~ | ~~Update human readable format in dashboard panels~~ | ~~Critical~~ | ~~Open~~ | ~~Kibana team will support human readable format in lens after that we can take this up: - [link](https://github.com/elastic/kibana/issues/153378)~~ | ~~[Phase-1](https://github.com/elastic/integrations/issues/5221)~~
~~Redis - #7114~~  | ~~Enhancement~~ | ~~Update human readable format in dashboard panels~~ | ~~Critical~~ | ~~Open~~ | ~~Kibana team will support human readable format in lens after that we can take this up: - [link](https://github.com/elastic/kibana/issues/153378)~~ | ~~[Phase-1](https://github.com/elastic/integrations/issues/5221)~~
~~Apache - #7108~~  | ~~Enhancement~~ | ~~Migrate Uptime visualization to the lens~~ | ~~Critical~~ | ~~Open~~ | ~~Kibana team will support human readable format in Kibana 8.10 lens after that we can take this up: - [link](https://github.com/elastic/kibana/issues/153378)~~ | ~~[Phase-1](https://github.com/elastic/integrations/issues/5221)~~
~GCP Load balancing - https://github.com/elastic/integrations/issues/7250~ | ~Enhancement~ | ~Migrate visualization to lens~ | ~Critical~ | ~Open~ | ~Blocked on incorrect field in dashboard #7242~  | ~[Phase 3](https://github.com/elastic/integrations/issues/6907)~
~Azure Metrics - #7158~  | ~Enhancement~ | ~Not correct metrics field~ | ~Major~ | ~Closed~ | ~Update fields in dashboard~ | ~[Phase 3](https://github.com/elastic/integrations/issues/6907)~
~MongoDB - #5846~ | ~Enhancement~ | ~Update MongoDB dashboards and visualizations~ | ~Minor~ | ~Open~ | ~Need to revisit this issue as per this [comment](https://github.com/elastic/integrations/pull/5831#issuecomment-1502796570)~ | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~AWS Metrics - #6338~ | ~Bug~ | ~Incorrect fields in AWS Metrics Overview dashboard~ | ~Critical~ | ~Closed~ | - | ~[Phase-2](https://github.com/elastic/integrations/issues/6141)~
~CoreDNS - #5518~ | ~Bug~ | ~CoreDNS dashboard visualizations are showing error~ | ~Critical~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~InfluxDB - #5506~ | ~Bug~ | ~The InfluxDB dashboard panel contains an error.~ | ~Critical~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~SpringBoot - #5871~ | ~Enhancement~ | ~SpringBoot dashboards contains meta field (_id)~ | ~Critical~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~Salesforce - #5876~ | ~Enhancement~ | ~Salesforce dashboards contains meta field (_id)~ | ~Critical~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~Cassandra - #5531~ | ~Enhancement~ | ~Visualizations with multiple breakdowns could not be migrated to lens~ | ~Critical~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~Redis - #5671~ | ~Bug~ | ~Need to update label position in Keyspaces [Metrics Redis] visualization~ | ~Minor~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~Redis - #5773~ | ~Enhancement~ | ~Need to increase the size of the pie chart~ | ~Minor~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~Hadoop - #5628~ | ~Bug~ | ~Remove unnecessary (s) from visualizations for Application dashboard~ | ~Minor~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~Vsphere - #5820~ | ~Enhancement~ | ~Change format of value to bytes~ | ~Minor~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~Postgres - #5797~ | ~Enhancement~ | ~Add label in Y-axis for Database Overview dashboard panels~ | ~Minor~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~IIS - #5796~ | ~Enhancement~ | ~Format value in Bytes in Bytes transferred/sec panel~ | ~Major~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~Nginx - #5795~ | ~Enhancement~ | ~Add Y-axis labels and Readjust column width in saved search of Nginx dashboards~ | ~Minor~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~Redis - #5723~ | ~Bug~ | ~Add description in Lens Migration~ | ~Minor~ | ~Closed~ | - | ~[Phase-1](https://github.com/elastic/integrations/issues/5221)~
~~STAN  - #7196~~ | ~~BugFix~~ | ~~Could not find reference error in STAN Metrics overview dashboard~~ | ~~Critical~~ | ~~Open~~ | ~~Blocked STAN dashboard migration (Resolve the error before migration)~~ | ~~[Phase 4](https://github.com/elastic/integrations/issues/7797)~~

</body>
</html>

**Comment by rameshelastic** [comment]
cc: @lalit-satapathy FYI

**Comment by agithomas** [comment]
@milan-elastic , please note that as part of the TSDB migration, you would see many packages promoted to using 8.8 as min version. I anticipate some of the pending issues can be unblocked with this

**Comment by lalit-satapathy** [comment]
> cc: @lalit-satapathy FYI

@rameshelastic need owner of this Meta issue.

**Comment by milan-elastic** [comment]
> @milan-elastic , please note that as part of the TSDB migration, you would see many packages promoted to using 8.8 as min version. I anticipate some of the pending issues can be unblocked with this

Sure, Let me check if we can work upon any backlog items now!
